Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -37,27 +43,134 @@ YUI.add('ez-roleserversideview', function (Y) {
         /**
          * tap event handler on the role assign buttons. It launches the
          * universal discovery widget so that the user can pick some contents.
+         * It also fill the config with datas concerning the section limitation
+         *
+         * @method _pickSubtreeWithSectionLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickSubtreeWithSectionLimitation: function (e) {
+            var button = e.target,
+                container = this.get('container'),
+                sectionSelector = container.one(".ez-role-assignment-section-id"),
+                sectionSelectedIndex = sectionSelector.get('selectedIndex'),
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                section = {
+                    sectionId: sectionSelector.get('options').item(sectionSelectedIndex).get('value'),
+                    sectionRestId: sectionSelector.get('options').item(sectionSelectedIndex).getAttribute('data-section-rest-id'),
+                    sectionName: sectionSelector.get('options').item(sectionSelectedIndex).get('text')
+                },
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
+                    roleName: button.getAttribute('data-role-name'),
+                    afterUpdateCallback: unsetLoading,
+                    limitationType: 'Section',
+                    section: section,
+                    sectionId: sectionSelector.get('options').item(sectionSelectedIndex).get('value'),
+                    sectionRestId: sectionSelector.get('options').item(sectionSelectedIndex).getAttribute('data-section-rest-id'),
+                    sectionName: sectionSelector.get('options').item(sectionSelectedIndex).get('text'),
+                };
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this._fireContentDiscover(button, unsetLoading, udwConfigData);
+        },
+
+        /**
+         * tap event handler on the role assign buttons. It launches the
+         * universal discovery widget so that the user can pick some contents.
          *
          * @method _pickSubtree
          * @protected
          * @param {EventFacade} e
          */
         _pickSubtree: function (e) {
             var button = e.target,
-                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button);
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
```

Comment:
`'data-role-rest-id'` and 'data-role-name' could be stored in a variable so you could avoid duplicate the strings in the module.